Library build fails due to ENR-related errors

### Describe the bug

When using Reth 1.0.0-rc.1 as a dependency, compilation of `reth-network-peers` fails.

### Steps to reproduce

```
$ cat src/main.rs
use std::sync::Arc;

use clap::Parser;
use reth::{
    builder::{InitState, NodeHandle, WithLaunchContext},
    cli::Cli,
    commands::node::NoArgs,
};
use reth_db::DatabaseEnv;
use reth_node_ethereum::EthereumNode;

fn main() -> eyre::Result<()> {
    Cli::<NoArgs>::parse().run(
        async move |builder: WithLaunchContext<Arc<DatabaseEnv>, InitState>, _args| {
            let NodeHandle {
                node,
                node_exit_future,
            } = builder.node(EthereumNode::default()).launch().await?;
            node.task_executor.spawn(Box::pin(async move {
                println!("Hello, world!");
            }));

            node_exit_future.await
        },
    )?;

    Ok(())
}
$ cat Cargo.toml
[package]
name = "rethenr"
version = "0.1.0"
edition = "2021"

[dependencies]
reth = { git = "https://github.com/paradigmxyz/reth", rev = "560080e" } # 1.0.0-rc.1
reth-db = { git = "https://github.com/paradigmxyz/reth", rev = "560080e" } # 1.0.0-rc.1
reth-node-ethereum = { git = "https://github.com/paradigmxyz/reth", rev = "560080e" } # 1.0.0-rc.1

[patch.crates-io]
revm = { git = "https://github.com/bluealloy/revm.git", rev = "41e2f7f" }
revm-interpreter = { git = "https://github.com/bluealloy/revm.git", rev = "41e2f7f" }
revm-precompile = { git = "https://github.com/bluealloy/revm.git", rev = "41e2f7f" }
revm-primitives = { git = "https://github.com/bluealloy/revm.git", rev = "41e2f7f" }
$ cargo run
error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:105:9
    |
105 |     Enr(Enr<secp256k1::SecretKey>),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:151:6
    |
151 | impl From<Enr<secp256k1::SecretKey>> for AnyNode {
    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:152:20
    |
152 |     fn from(value: Enr<secp256k1::SecretKey>) -> Self {
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:176:6
    |
176 | impl TryFrom<&Enr<secp256k1::SecretKey>> for NodeRecord {
    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:179:23
    |
179 |     fn try_from(enr: &Enr<secp256k1::SecretKey>) -> Result<Self, Self::Error> {
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`

error[E0599]: the method `ip4` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:180:33
    |
180 |         let Some(address) = enr.ip4().map(IpAddr::from).or_else(|| enr.ip6().map(IpAddr::from))
    |                                 ^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `ip6` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:180:72
    |
180 |         let Some(address) = enr.ip4().map(IpAddr::from).or_else(|| enr.ip6().map(IpAddr::from))
    |                                                                        ^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `udp4` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:185:34
    |
185 |         let Some(udp_port) = enr.udp4().or_else(|| enr.udp6()) else {
    |                                  ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `udp6` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:185:56
    |
185 |         let Some(udp_port) = enr.udp4().or_else(|| enr.udp6()) else {
    |                                                        ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `tcp4` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:189:34
    |
189 |         let Some(tcp_port) = enr.tcp4().or_else(|| enr.tcp6()) else {
    |                                  ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `tcp6` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:189:56
    |
189 |         let Some(tcp_port) = enr.tcp4().or_else(|| enr.tcp6()) else {
    |                                                        ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `public_key` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/node_record.rs:193:36
    |
193 |         let id = crate::pk2id(&enr.public_key());
    |                                    ^^^^^^^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:105:9
    |
98  |     Debug, Clone, Eq, PartialEq, Hash, serde_with::SerializeDisplay, serde_with::DeserializeFromStr,
    |     ----- in this derive macro expansion
...
105 |     Enr(Enr<secp256k1::SecretKey>),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`, which is required by `&Enr<secp256k1::SecretKey>: Debug`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
    = note: required for `Enr<secp256k1::SecretKey>` to implement `Debug`
    = note: 1 redundant requirement hidden
    = note: required for `&Enr<secp256k1::SecretKey>` to implement `Debug`
    = note: required for the cast from `&&Enr<secp256k1::SecretKey>` to `&dyn Debug`
    = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:105:9
    |
98  |     Debug, Clone, Eq, PartialEq, Hash, serde_with::SerializeDisplay, serde_with::DeserializeFromStr,
    |            ----- in this derive macro expansion
...
105 |     Enr(Enr<secp256k1::SecretKey>),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`
    = note: this error originates in the derive macro `Clone` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:105:9
    |
98  |     Debug, Clone, Eq, PartialEq, Hash, serde_with::SerializeDisplay, serde_with::DeserializeFromStr,
    |                   -- in this derive macro expansion
...
105 |     Enr(Enr<secp256k1::SecretKey>),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`
    = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0369]: binary operation `==` cannot be applied to type `Enr<secp256k1::SecretKey>`
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:105:9
    |
98  |     Debug, Clone, Eq, PartialEq, Hash, serde_with::SerializeDisplay, serde_with::DeserializeFromStr,
    |                       --------- in this derive macro expansion
...
105 |     Enr(Enr<secp256k1::SecretKey>),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in the derive macro `PartialEq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:105:9
    |
98  |     Debug, Clone, Eq, PartialEq, Hash, serde_with::SerializeDisplay, serde_with::DeserializeFromStr,
    |                                  ---- in this derive macro expansion
...
105 |     Enr(Enr<secp256k1::SecretKey>),
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`, which is required by `Enr<secp256k1::SecretKey>: std::hash::Hash`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
    = note: required for `Enr<secp256k1::SecretKey>` to implement `std::hash::Hash`
    = note: this error originates in the derive macro `Hash` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: the method `public_key` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:116:42
    |
116 |             Self::Enr(enr) => pk2id(&enr.public_key()),
    |                                          ^^^^^^^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `ip4` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:129:26
    |
128 |                       address: enr
    |  ______________________________-
129 | |                         .ip4()
    | |                         -^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    | |_________________________|
    | 
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  |   pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    |   -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `ip6` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:131:41
    |
131 |                         .or_else(|| enr.ip6().map(std::net::IpAddr::from))?,
    |                                         ^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `tcp4` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:132:35
    |
132 |                     tcp_port: enr.tcp4().or_else(|| enr.tcp6())?,
    |                                   ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `tcp6` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:132:57
    |
132 |                     tcp_port: enr.tcp4().or_else(|| enr.tcp6())?,
    |                                                         ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `udp4` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:133:35
    |
133 |                     udp_port: enr.udp4().or_else(|| enr.udp6())?,
    |                                   ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `udp6` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:133:57
    |
133 |                     udp_port: enr.udp4().or_else(|| enr.udp6())?,
    |                                                         ^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0599]: the method `public_key` exists for reference `&Enr<SecretKey>`, but its trait bounds were not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:134:36
    |
134 |                     id: pk2id(&enr.public_key()),
    |                                    ^^^^^^^^^^ method cannot be called on `&Enr<SecretKey>` due to unsatisfied trait bounds
    |
   ::: /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/secp256k1-0.28.2/src/key.rs:58:1
    |
58  | pub struct SecretKey([u8; constants::SECRET_KEY_SIZE]);
    | -------------------- doesn't satisfy `secp256k1::SecretKey: EnrKey`
    |
    = note: the following trait bounds were not satisfied:
            `secp256k1::SecretKey: EnrKey`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:153:19
    |
153 |         Self::Enr(value)
    |                   ^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:173:20
    |
173 |             return Enr::from_str(s).map(AnyNode::Enr)
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:173:20
    |
173 |             return Enr::from_str(s).map(AnyNode::Enr)
    |                    ^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`, which is required by `Enr<_>: FromStr`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
    = note: required for `Enr<secp256k1::SecretKey>` to implement `FromStr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:173:20
    |
173 |             return Enr::from_str(s).map(AnyNode::Enr)
    |                    ^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`, which is required by `Enr<secp256k1::SecretKey>: FromStr`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
    = note: required for `Enr<secp256k1::SecretKey>` to implement `FromStr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /home/jmcph4/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:184:42
    |
184 |             Self::Enr(enr) => write!(f, "{enr}"),
    |                                          ^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /home/jmcph4/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`
    = note: this error originates in the macro `$crate::format_args` which comes from the expansion of the macro `write` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0277, E0369, E0599.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `reth-network-peers` (lib) due to 30 previous errors
$ echo $?
101
```

### Node logs

```text
N/A
```


### Platform(s)

Linux (x86)

### What version/commit are you on?

`560080e` (1.0.0-rc.1)

### What database version are you on?

N/A

### Which chain / network are you on?

N/A

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

N/A

### If you've built Reth from source, provide the full command you used

`cargo run`

### Code of Conduct

- [X] I agree to follow the Code of Conduct

I am having similar issues:

```
error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /Users/me/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:173:20
    |
173 |             return Enr::from_str(s).map(AnyNode::Enr)
    |                    ^^^^^^^^^^^^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`, which is required by `Enr<secp256k1::SecretKey>: FromStr`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
    = note: required for `Enr<secp256k1::SecretKey>` to implement `FromStr`

error[E0277]: the trait bound `secp256k1::SecretKey: EnrKey` is not satisfied
   --> /Users/me/.cargo/git/checkouts/reth-36d3ea1d1152b20c/560080e/crates/net/peers/src/lib.rs:184:42
    |
184 |             Self::Enr(enr) => write!(f, "{enr}"),
    |                                          ^^^^^ the trait `EnrKey` is not implemented for `secp256k1::SecretKey`
    |
    = help: the following other types implement trait `EnrKey`:
              CombinedKey
              ecdsa::signing::SigningKey<enr::k256::Secp256k1>
              enr::ed25519_dalek::SigningKey
              enr::secp256k1::SecretKey
note: required by a bound in `Enr`
   --> /Users/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/enr-0.12.1/src/lib.rs:228:19
    |
228 | pub struct Enr<K: EnrKey> {
    |                   ^^^^^^ required by this bound in `Enr`
    = note: this error originates in the macro `$crate::format_args` which comes from the expansion of the macro `write` (in Nightly builds, run with -Z macro-backtrace for more info)

--- a/crates/net/peers/Cargo.toml
+++ b/crates/net/peers/Cargo.toml
@@ -34,4 +34,4 @@ secp256k1 = { workspace = true, features = ["rand"] }
 serde_json.workspace = true
 
 [features]
-secp256k1 = ["dep:secp256k1"]
+secp256k1 = ["dep:secp256k1", "enr/secp256k1"]

